cpp/distance_calculator/distance_calculator.cpp

```diff
@@ -0,0 +1,136 @@
+    const mgp::Node& node1 = arguments[0].ValueNode();
+    const mgp::Node& node2 = arguments[1].ValueNode();
+    const std::string_view metrics = arguments[2].ValueString();
+    int64_t decimals = arguments[3].ValueInt();
+
+    mgp::Record record = record_factory.NewRecord();
+
+    record.Insert(kReturnDistance, distance_calc(node1, node2, metrics=="km", decimals));
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+
+}
+
+void Multiple(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try{
+    const auto arguments = mgp::List(args);
+    mgp::List distances= mgp::List();
+
+    const mgp::List start_points = arguments[0].ValueList();
+    const mgp::List end_points = arguments[1].ValueList();
+    const std::string_view metrics = arguments[2].ValueString();
+    int64_t decimals = arguments[3].ValueInt();
+
+    if (start_points.Size() != end_points.Size()){
+      throw mgp::ValueException("Length of both arrays must be of equal size.");
+    }
+    bool use_km =  metrics=="km";
+    for(int i=0; i< start_points.Size(); i++){
+      const mgp::Node& node1 = start_points[i].ValueNode();
+      const mgp::Node& node2 = end_points[i].ValueNode();
+      distances.AppendExtend(mgp::Value(distance_calc(node1, node2, use_km, decimals)));
+    }
+
+    mgp::Record record = record_factory.NewRecord();
+    record.Insert(kReturnDistances, distances);
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+  
+}
+
+
+
+extern "C" int mgp_init_module(struct mgp_module *module, struct mgp_memory *memory) {
+  try {
+    mgp::memory = memory;
+
+    const char* metrics = "m";
+
+    AddProcedure(Single, kProcedureSingle, mgp::ProcedureType::Read,
+                 {mgp::Parameter(kArgumentStartPoint, mgp::Type::Node), 
+                 mgp::Parameter(kArgumentEndPoint, mgp::Type::Node),
+                 mgp::Parameter(kArgumentMetrics, mgp::Type::String, metrics),
+                 mgp::Parameter(kArgumentDecimals, mgp::Type::Int, static_cast<int64_t>(2))}, 
+                 {mgp::Return(kReturnDistance, mgp::Type::Double)},
+                 module, memory);
+
+    const auto multiple_input = std::make_pair(mgp::Type::List, mgp::Type::Node);
+    const auto multiple_return = std::make_pair(mgp::Type::List, mgp::Type::Double);
+    AddProcedure(Multiple, kProcedureMultiple, mgp::ProcedureType::Read,
+                 {mgp::Parameter(kArgumentStart, multiple_input), mgp::Parameter(kArgumentEnd, multiple_input), 
+                  mgp::Parameter(kArgumentMetrics, mgp::Type::String, metrics),
+                  mgp::Parameter(kArgumentDecimals, mgp::Type::Int, static_cast<int64_t>(2))}, 
+                 {mgp::Return(kReturnDistances, multiple_return)},
+                 module, memory);
```

Comment:
Extract `"m"` and `2` as constants; I also think you won’t need static cast anymore with that.

Reply:
Yes. Good catch.